jscomp/others/js_null_undefined.ml

```diff
@@ -33,11 +33,11 @@ external empty : 'a t = "undefined" [@@bs.val]
 
 let bind x f =
   match to_opt x with
-  | None ->  undefined
-  | Some x -> return (f  x [@bs])
+  | None -> Obj.magic x
```

Comment:
can you write type annotations when using Obj.magic? otherwise it is unclear to me what you re doing here
```
(Obj.magic (x : 'a ) : 'b)
```
do you think it makes sense to provide a sugar like this `[%bs.unsafe_cat: 'a -> 'b ]`